File: src/contests/cg/puzzle/cpp/euclids-algorithm.cpp
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

#include <iostream>
using namespace std;

int main() {
    long long a, b;
    cin >> a >> b;
    long long x = a, y = b;
    while (y != 0) {
        long long q = x / y, r = x % y;
        cout << x << "=" << y << "*" << q << "+" << r << "\n";
        x = y;
        y = r;
    }
    cout << "GCD(" << a << "," << b << ")=" << x << "\n";
}

File: src/contests/cg/puzzle/pascal/euclids-algorithm.pas
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

{$mode objfpc}{$H+}
var
  a, b, x, y, q, r: int64;
begin
  readln(a, b);
  x := a;
  y := b;
  while y <> 0 do
  begin
    q := x div y;
    r := x mod y;
    writeln(x, '=', y, '*', q, '+', r);
    x := y;
    y := r;
  end;
  writeln('GCD(', a, ',', b, ')=', x);
end.
